### PR TITLE
Update s3fs_util.cpp for correspondence of Nextcloud contype

### PR DESCRIPTION
**Pull request by LutzFinsterle2019**
Nextcloud Compatibility in directory mime-types

### Relevant Issue (if applicable)
Compatibility with nextcloud directory mimetypes

### Details
Access Nextcloud used buckets with s3fs

### CODE REVIEW

File: src/s3fs_util.cpp [hunk 1]
```diff
@@ -814,7 +814,7 @@ mode_t get_mode(headers_t& meta, const char* path, bool checkdir, bool forcedir)
             if(string::npos != pos){
               strConType = strConType.substr(0, pos);
             }
-            if(strConType == "application/x-directory"){
+            if(strConType == "application/x-directory" || strConType == "httpd/unix-directory"){
```
Comment (gaul) [inline src/s3fs_util.cpp]: Could you add a comment explaining what this constant means?

Comment (LutzFinsterle2019) [reply src/s3fs_util.cpp]: Sorry for answering late, have been busy lately.
The comment is: "Nextcloud stores Directory objects with this mime type when mounting a Bucket as external Storage"
I have created a new pull for introducing the Comment

Comment (LutzFinsterle2019) [reply src/s3fs_util.cpp]: Sorry for answering late, have been busy lately.
The comment is: "Nextcloud stores Directory objects with this mime type when mounting a Bucket as external Storage"

File: src/s3fs_util.cpp [hunk 2]
```diff
@@ -814,7 +814,8 @@ mode_t get_mode(headers_t& meta, const char* path, bool checkdir, bool forcedir)
             if(string::npos != pos){
               strConType = strConType.substr(0, pos);
             }
-            if(strConType == "application/x-directory"){
+            if(strConType == "application/x-directory" 
+	       || strConType == "httpd/unix-directory"){ # Nextcloud uses this MIME type for directory objects when mounting bucket as external Storage
```
Comment (gaul) [inline src/s3fs_util.cpp]: C++ uses `//` for comments.

Comment (LutzFinsterle2019) [reply src/s3fs_util.cpp]: Sorry :-) was just too fast

**Comment by gaul** [comment]
Also when squashing can you add a more descriptive commit message?

**Comment by ggtakec** [comment]
@gaul Since PR code is no problem, I added intention to PR title instead of descriptive commit message.
Also, I wanted the commit message to be compressed, but since there is no progress, I will merge as it is.

@LutzFinsterle2019 Thanks for your PR.

**Pull request merged**